dbms/src/Storages/DeltaMerge/Index/LocalIndexWriter.h

```diff
@@ -0,0 +1,55 @@
+// Copyright 2025 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include <Columns/ColumnVector.h>
+#include <Columns/IColumn.h>
+#include <DataTypes/IDataType.h>
+#include <Storages/DeltaMerge/Index/LocalIndexInfo.h>
+#include <Storages/DeltaMerge/Index/LocalIndexWriter_fwd.h>
+#include <Storages/DeltaMerge/dtpb/column_file.pb.h>
+#include <Storages/DeltaMerge/dtpb/dmfile.pb.h>
+#include <Storages/Page/PageDefinesBase.h>
+
+
+namespace DB::DM
+{
+
+/// Builds a LocalIndex.
+class LocalIndexWriter
+{
+public:
+    using ProceedCheckFn = std::function<bool()>;
+
+public:
+    explicit LocalIndexWriter(const LocalIndexInfo & index_info)
+        : index_info(index_info)
+    {}
+
+    static LocalIndexWriterPtr create(const LocalIndexInfo & index_info, bool in_memory);
```

Comment:
This is not compatible with FTS, in FTS OnDisk Writer must specify a disk path at construction, because it is OnDisk.